irmaclient/logs.go

```diff
@@ -35,6 +36,72 @@ type LogEntry struct {
 	request    irma.SessionRequest   // cached parsed version of Request; get with LogEntry.SessionRequest()
 }
 
+// We need manual unmarshalling to deal with legacy log entries that have
+func (entry *LogEntry) UnmarshalJSON(b []byte) (err error) {
+	// Internal type needed for unmarshalling to delay requestorInfo unmarshalling
+	// NOTE: keep this up to date with above definition of log entry!
+	var temp struct {
+		// General info
+		ID   uint64
+		Type irma.Action
+		Time irma.Timestamp // Time at which the session was completed
+
+		// Credential removal
+		Removed map[irma.CredentialTypeIdentifier][]irma.TranslatedString `json:",omitempty"`
+
+		// Signature sessions
+		SignedMessage          []byte          `json:",omitempty"`
+		Timestamp              *atum.Timestamp `json:",omitempty"`
+		SignedMessageLDContext string          `json:",omitempty"`
+
+		// Issuance sessions
+		IssueCommitment *irma.IssueCommitmentMessage `json:",omitempty"`
+
+		// All session types
+		ServerName *json.RawMessage      `json:",omitempty"`
```

Comment:
Why does this have to be a pointer to json.RawMessage? In `Request` it is not a pointer. When it is not a pointer and it is empty it will just get the value of an empty array and then you can distinguish it too right? Or is the empty message also a specific ServerName?

Reply:
Per the programmed old format, ServerName can be absent in json logenties. For these cases, the canonical representation with a RequestorInfo is a nilpointer to one, not an empty RequestorInfo. This just ensures that that case can be easily distinguished in the code following.

Reply:
Okay, because it is a temporary type it is acceptable then. The full class is still a bit verbose of course (that's why all the manual assignments are needed. Since this is code that cannot be removed at some point (legacy logs will always be there), did you consider parsing this in a `UnmashalJSON` function for `RequestorInfo` instead of doing it in the `UnmarshalJSON` of `LogEntry`? Then you would not need all the manual assignments if I'm right.

Reply:
Did consider it, but did not go for it as this is where the actual incompatibility arises from, which makes it semantically easier to reason about when fixed here. This also prevents any issues with RequestorInfo being misparsed from schemes, where TranslatedValues are not acceptable.

Reply:
So you then mean that this is not the only place where `RequestorInfo` is parsed and at the other places it is actually weird to include the legacy parsing too so if I understand you right?

Reply:
Yes, RequestorInfo is also parsed directly from the RequestorSchemes, and there legacy parsing is unwanted

Reply:
I see, then that's no option indeed. Then it is fine this way I think.